src/els_uri.erl

```diff
@@ -30,9 +30,47 @@ module(Uri) ->
   binary_to_atom(filename:basename(path(Uri), <<".erl">>), utf8).
 
 -spec path(uri()) -> path().
-path(<<"file://", Path/binary>>) ->
-  Path.
+path(Uri) ->
+  Uri1 = http_uri:decode(Uri),
+  #{
+    host := Host,
+    path := Path,
+    scheme := <<"file">>
+  } = uri_string:parse(Uri1),
+
+  case {is_windows(), is_local(Host)} of
+    {_, true} ->
+      string:slice(Path, 1);
+    {true, false} ->
+      <<"//", Host/binary, Path/binary>>;
+    {false, false} ->
+      error(badarg)
+  end.
 
 -spec uri(path()) -> uri().
 uri(Path) ->
-  <<"file://", Path/binary>>.
+  Path1 =
+  case is_windows() of
+    true ->
+      list_to_binary(string:replace(Path, <<"\\">>, <<"/">>));
+    _ ->
+      Path
+  end,
+  <<"file:///", Path1/binary>>.
+
+-spec is_windows() -> boolean().
+is_windows() ->
+  case os:type() of
+    {win32, _} ->
+      true;
+    _ ->
+      false
+  end.
+
+-spec is_local(binary()) -> boolean().
+is_local(<<>>) ->
+  true;
+is_local(<<"localhost">>) ->
+  true;
+is_local(_) ->
+  false.
```

Comment:
Optional: A more concise alternative.
```suggestion
-spec is_local(binary()) -> boolean().
is_local(Host) ->
  Host =:= <<>> orelse Host =:= <<"localhost">>.
```

Reply:
I'll apply this one as well and squash the commits.